### ISSUE TITLE
Add automatic pre and post events

### ISSUE THREAD

**Issue by da-h** [da-h/miniflask#15]
**Description**
It would be a nice addition to preprocess/postprocess any event easily.
For every event, say `myevent`, miniflask could check wether any pre/post events exist.

**API Proposals:**
- `_myevent` and `myevent_` (underscore defines if the event shall be executed before/after `myevent`
- `myevent_pre` and `myevent_post`
- `myevent_before` and `myevent_after`
- `before_myevent` and `after_myevent`
 
**Note**:
It would be easy to just add optional calls to pre and post events. However, the additional two ifs can be omitted, if the pre and post events are backed directly into the wrapper function. 

**Note**:
Also take multiple steps into consideration.

**Comment by da-h** [comment]
@sbrodehl: What is your take on this API-addition?

**Comment by da-h** [comment]
This is indeed possible.
However, the current API requires to just name the event for a definition.

E.g.:
```
def register(mf):
   mf.register_event('eventname`, lambda x: x)
```

It would be possible to define pre/post Events using dot-Notation, however I am not sure about the runtime overhead it would introduce. In that case all Events needed to be seperate Objects which is currently not the case due to performance reasons.

**Comment by da-h** [comment]
Implemented as of 266d523.
